src/main/java/com/synopsys/integration/jenkins/detect/service/strategy/DetectScriptStrategy.java

```diff
@@ -82,21 +83,21 @@ public DetectScriptStrategy(JenkinsIntLogger logger, JenkinsProxyHelper jenkinsP
     }
 
     @Override
-    public MasterToSlaveCallable<String, IntegrationException> getSetupCallable() {
-        String scriptUrl;
+    public MasterToSlaveCallable<String, IntegrationException> getSetupCallable() throws IntegrationException {
+        HttpUrl httpUrl;
         String scriptFileName;
         if (operatingSystemType == OperatingSystemType.WINDOWS) {
-            scriptUrl = LATEST_POWERSHELL_SCRIPT_URL;
+            httpUrl = new HttpUrl(LATEST_POWERSHELL_SCRIPT_URL);
             scriptFileName = POWERSHELL_SCRIPT_FILENAME;
         } else {
-            scriptUrl = LATEST_SHELL_SCRIPT_URL;
+            httpUrl = new HttpUrl(LATEST_SHELL_SCRIPT_URL);
             scriptFileName = SHELL_SCRIPT_FILENAME;
         }
 
         // ProxyInfo itself isn't serializable, so we unpack it into serializable pieces and rebuild it later when we download the script. -- rotte JUL 2020
         ProxyInfo proxyInfo;
         try {
-            proxyInfo = jenkinsProxyHelper.getProxyInfo(scriptUrl);
+            proxyInfo = jenkinsProxyHelper.getProxyInfo(httpUrl.string());
```

Comment:
A note beyond the scope of this PR: JenkinsProxyHelper should take an HttpUrl-- HttpUrl would wrap the URL handling logic that we do in JenkinsProxyHelper::shouldNotUseProxy, simplifying the class.

Reply:
Ticket opened in Jira